### PR TITLE
TILA-2382 - Reservation refund mutation for admins

### PR DESCRIPTION
**Pull request by saaste**
## Change log
- Added mutation for admins to refund reservations

## Deployment reminder
- No changes required

### CODE REVIEW

File: api/graphql/reservations/reservation_serializers/refund_serializers.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+from api.graphql.validation_errors import ValidationErrorCodes, ValidationErrorWithCode
+from merchants.models import OrderStatus, PaymentOrder
+from reservations.models import STATE_CHOICES, Reservation
+from reservations.tasks import refund_paid_reservation_task
+
+DEFAULT_TIMEZONE = get_default_timezone()
+
+
+class ReservationRefundSerializer(PrimaryKeySerializer):
+    def __init__(self, *args, **kwargs):
```
Comment (nicobav) [inline api/graphql/reservations/reservation_serializers/refund_serializers.py:16]: Perhaps some leftover from testing(besthing)?

File: api/graphql/reservations/reservation_serializers/refund_serializers.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        super().__init__(*args, **kwargs)
+
+    class Meta:
+        model = Reservation
+        fields = [
+            "pk",
+        ]
+
+    @property
+    def validated_data(self):
```
Comment (nicobav) [inline api/graphql/reservations/reservation_serializers/refund_serializers.py:26]: Same as some; leftover from testing this is?

File: api/graphql/reservations/reservation_serializers/refund_serializers.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        validated_data = super().validated_data
+        return validated_data
+
+    def validate(self, data):
+        now = datetime.datetime.now(tz=DEFAULT_TIMEZONE)
+        if self.instance.price_net <= Decimal("0.0"):
+            raise ValidationErrorWithCode(
+                "Only reservations with price > 0 can be refunded.",
```
Comment (nicobav) [inline api/graphql/reservations/reservation_serializers/refund_serializers.py:34]: I'd change the `price > 0` to `price greater than zero`

File: users/utils/open_city_profile/basic_info_resolver.py [hunk 4]
```diff
@@ -99,6 +101,7 @@ def __read_fields(self):
             settings.OPEN_CITY_PROFILE_GRAPHQL_API,
             json={"query": query},
             headers={"Authorization": self.token},
+            timeout=REQUEST_TIMEOUT_SECONDS,
```
Comment (nicobav) [inline users/utils/open_city_profile/basic_info_resolver.py:104]: ❤️

File: users/utils/open_city_profile/birthday_resolver.py [hunk 5]
```diff
@@ -156,6 +158,7 @@ def __make_profile_request(self, token) -> dict:
             settings.OPEN_CITY_PROFILE_GRAPHQL_API,
             json={"query": query},
             headers={"Authorization": token},
+            timeout=REQUEST_TIMEOUT_SECONDS,
```
Comment (nicobav) [inline users/utils/open_city_profile/birthday_resolver.py:161]: ❤️

**Review by nicobav — CHANGES_REQUESTED** [review]
Grreat! Just some brooming left to do 🧹

File: api/graphql/reservations/reservation_serializers/refund_serializers.py (hunk 1 above)
Comment (saaste) [reply api/graphql/reservations/reservation_serializers/refund_serializers.py]: I believe this is a doing of copypaste monster 😇
I'll remove these pointless override functions 🧹

File: api/graphql/reservations/reservation_serializers/refund_serializers.py (hunk 3 above)
Comment (saaste) [reply api/graphql/reservations/reservation_serializers/refund_serializers.py]: True, that is more human and less nerdy 🤓

**Review by nicobav — APPROVED** [review]

**Pull request merged**